flink-runtime/src/test/java/org/apache/flink/runtime/blob/BlobClientTest.java

```diff
@@ -487,4 +488,58 @@ private static void uploadJarFile(
 			validateGetAndClose(blobClient.getInternal(jobId, blobKeys.get(0)), testFile);
 		}
 	}
+
+
+	/**
+	 * Tests the socket operation timeout.
+	 */
+	@Test
+	public void testSocketTimeout() {
+		Configuration clientConfig = getBlobClientConfig();
+		int oldSoTimeout = clientConfig.getInteger(BlobServerOptions.SO_TIMEOUT);
+
+		clientConfig.setInteger(BlobServerOptions.SO_TIMEOUT, 50);
+		getBlobServer().setBlockingMillis(10_000);
+
+		try {
+			InetSocketAddress serverAddress = new InetSocketAddress("localhost", getBlobServer().getPort());
+
+			try (BlobClient client = new BlobClient(serverAddress, clientConfig)) {
+				client.getInternal(new JobID(), BlobKey.createKey(TRANSIENT_BLOB));
+
+				fail("Should throw a exception.");
+			} catch (Throwable t) {
+				assertEquals(java.net.SocketTimeoutException.class, ExceptionUtils.stripException(t, IOException.class).getClass());
+			}
+		} finally {
+			clientConfig.setInteger(BlobServerOptions.SO_TIMEOUT, oldSoTimeout);
```

Comment:
Seems we reuse the blobserver for all the tests here, Will we change to each blobserver for each test by changing annotation of `@BeforeClass` to `@Before` and `@AfterClass` to `@After`

Reply:
In that case, the blob server need to be started every time, which will lengthen the test time.